### PR TITLE
Add support for the Rust programming language

### PR DESCRIPTION
**Pull request by ndenev**
Adding support for the Rust programming language. Mostly based (read copy&paste) on the existing Python and Java implementations.

### CODE REVIEW

**Comment by ndenev** [comment]
I'm sorry for the multiple commits, but for some reason I wasn't able to make all the tests work on my machine so I've missed a few. (Also looks like some of the existing tests print some non-printable chars that really screw my terminal when running them).

File: test/org/opensolaris/opengrok/search/SearchTest.java [hunk 1]
```diff
@@ -149,35 +149,35 @@ public void testSearch() {
```
Comment (tulinkry) [inline test/org/opensolaris/opengrok/search/SearchTest.java]: Incorrect copyright -> 2016

File: src/org/opensolaris/opengrok/analysis/rust/RustAnalyzer.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+package org.opensolaris.opengrok.analysis.rust;
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/analysis/rust/RustAnalyzer.java]: missing copyright
for example
```java
/*
 * Copyright (c) 2016, Oracle and/or its affiliates. All rights reserved.
 */
package org.opensolaris.opengrok.configuration.messages;
```

File: src/org/opensolaris/opengrok/analysis/rust/RustAnalyzerFactory.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+package org.opensolaris.opengrok.analysis.rust;
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/analysis/rust/RustAnalyzerFactory.java]: missing copyright

File: src/org/opensolaris/opengrok/analysis/rust/RustSymbolTokenizer.lex [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/analysis/rust/RustSymbolTokenizer.lex]: missing copyright

File: src/org/opensolaris/opengrok/analysis/rust/RustXref.lex [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/analysis/rust/RustXref.lex]: missing copyright

**Comment by tarzanek** [comment]
@ndenev  thank you very much for your changes, besides Krystofs comments there is one administrative thing - we will need you to either agree with OCA here,
or check: http://www.oracle.com/technetwork/community/oca-486395.html - follow the submission procedure and you can ping me here, I should be able to internally check very quickly if your submission is on file (we don't need to wait once they update above page) - this path will make your next patches easily acceptable, since we will avoid the legal checks again

tia
L

**Comment by ndenev** [comment]
@tarzanek, thanks, pushed some changes addressing Krystofs comments (and squashed the remaining commints) couple of days ago.
I agree to the OCA and I hope I'll be able to submit this in the next few days when time permits.

Thanks.

**Comment by tarzanek** [comment]
Спасибо Nikolay ,
the agreement you wrote is OK, if you add paper work to it it will be even better, but merge can happen already

thnx
L

**Pull request merged**